packages/metals-vscode/src/util.ts

```diff
@@ -48,14 +50,29 @@ export function getConfigValue<A>(
 }
 
 export function metalsDir(target: ConfigurationTarget): string {
-  if (target == ConfigurationTarget.Workspace && workspace.workspaceFolders) {
-    const wsDir = workspace.workspaceFolders[0]?.uri.fsPath;
+  const workspaceFolder = currentWorkspaceFolder();
+  if (target == ConfigurationTarget.Workspace && workspaceFolder) {
+    const wsDir = workspaceFolder.uri.fsPath;
     return path.join(wsDir, ".metals");
   } else {
     return path.join(os.homedir(), ".metals");
   }
 }
 
+export function currentWorkspaceFolder(): WorkspaceFolder | undefined {
+  const activeEditor = window.activeTextEditor;
+  let workspaceIndex = 0;
+  if (activeEditor) {
+    const workspaceFolder = workspace.getWorkspaceFolder(
+      activeEditor.document.uri
+    );
+    if (workspaceFolder) {
+      workspaceIndex = workspaceFolder.index;
+    }
+  }
+  return workspace.workspaceFolders?.[workspaceIndex];
+}
```

Comment:
Possibly a bit easier to read?

```suggestion
	const activeEditor = window.activeTextEditor;
  if (activeEditor) {
  	return workspace.getWorkspaceFolder(activeEditor.document.uri)
	}
	return workspace.workspaceFolders?.[0];
}
```

There's a minor difference in logic but I don't know if it's practically a difference: if `getWorkspaceFolder` returns `undefined` we return `undefined` instead of `workspaceFolders?.[0]`.

Reply:
++ for @gabro snippet. My 2c:
```suggestion
  const activeEditorUri = window.activeTextEditor?.document.uri;
  if (activeEditorUri) {
    return workspace.getWorkspaceFolder(activeEditorUri);
  }
  return workspace.workspaceFolders?.[0];
}
```

Reply:
Thanks! Fixed!